### ISSUE TITLE
[Regression] 2.142.0 messed up metadata folder structure

### ISSUE THREAD

**Issue by timc-mediamonks** [fastlane/fastlane#16050]
### New Regression Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Regression Information

- Breaking version: [e.g. `2.142.0`]
- Last working version: [e.g. `2.141.0`]

### Regression Description

Running `bundle exec fastlane screengrab`, with `locales(['en-US', 'de-DE', 'nl-NL'])` I expect the following folder structure to be generated

```
metadata/android/de-DE/images/phoneScreenshots/*.png
metadata/android/en-US/images/phoneScreenshots/*.png
metadata/android/nl-NL/images/phoneScreenshots/*.png
```
Instead I get
```
metadata/android/de-DE/images/de-DE/images/phoneScreenshots/*.png
metadata/android/de-DE/images/nl-NL/images/phoneScreenshots/*.png
metadata/android/en-US/images/de-DE/images/phoneScreenshots/*.png
metadata/android/en-US/images/en-US/images/phoneScreenshots/*.png
metadata/android/en-US/images/nl-NL/images/phoneScreenshots/*.png
metadata/android/nl-NL/images/nl-NL/images/phoneScreenshots/*.png
```

Adding more locales in Screengrabfile adds an increasing amount of excess folders.

This also breaks the screenshots.html. It just shows a blank screen. Seems to be caused by changes in PR https://github.com/fastlane/fastlane/pull/15994

##### Complete output when running fastlane, including the stack trace and command used

<details>
  <pre>➜  xxx git:(feature/fastlane-screenshots) ✗ bundle exec fastlane screengrab
[⠼] 🚀 /Users/timcastelijns/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/faraday_middleware-0.13.1/lib/faraday_middleware/response_middleware.rb:14: warning: Capturing the given block using Proc.new is deprecated; use `&block` instead
[✔] 🚀
[11:30:00]: Successfully loaded '/Users/timcastelijns/git/xxx/fastlane/Screengrabfile' 📄

+----------------------------+----------------------------------+
|       Detected Values from './fastlane/Screengrabfile'        |
+----------------------------+----------------------------------+
| locales                    | ["en-US", "de-DE", "nl-NL"]      |
| app_package_name           | com.xxx          |
| use_tests_in_packages      | ["com.xxx.screengrab"] |
| clear_previous_screenshots | true                             |
+----------------------------+----------------------------------+

+-----------------------------+------------------------------------------+
|                     Summary for screengrab 2.142.0                     |
+-----------------------------+------------------------------------------+
| tests_package_name          | com.xxx.debug.test             |
| android_home                | /Users/timcastelijns/Library/Android/sdk |
| locales                     | ["en-US", "de-DE", "nl-NL"]              |
| clear_previous_screenshots  | true                                     |
| output_directory            | fastlane/metadata/android                |
| skip_open_summary           | false                                    |
| app_package_name            | com.xxx.debug                  |
| use_tests_in_packages       | ["com.xxx.screengrab"]         |
| test_instrumentation_runner | androidx.test.runner.AndroidJUnitRunner  |
| ending_locale               | en-US                                    |
| use_adb_root                | false                                    |
| device_type                 | phone                                    |
| exit_on_test_failure        | true                                     |
| reinstall_app               | false                                    |
| use_timestamp_suffix        | true                                     |
+-----------------------------+------------------------------------------+

[11:30:00]: Clearing phoneScreenshots within fastlane/metadata/android
[11:30:00]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell echo \$EXTERNAL_STORAGE
[11:30:00]: ▸ /sdcard
[11:30:00]: Cleaning screenshots on device
[11:30:00]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell rm -rf /sdcard/com.xxx.debug/screengrab
ls: /data/data/com.xxx.debug/app_screengrab: Permission denied
[11:30:00]: Exit status: 1
/Users/timcastelijns/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/fastlane-2.142.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141: warning: Using the last argument as keyword parameters is deprecated; maybe ** should be added to the call
/Users/timcastelijns/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/fastlane-2.142.0/fastlane_core/lib/fastlane_core/ui/errors/fastlane_error.rb:9: warning: The called method `initialize' is defined here
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/en-US/images/screenshots: Permission denied
[11:30:00]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/de-DE/images/screenshots: Permission denied
[11:30:00]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/nl-NL/images/screenshots: Permission denied
[11:30:01]: Exit status: 1
[11:30:01]: To not be asked about this value, you can specify it using 'app_apk_path'
[11:30:01]: Select your debug app APK
/Users/timcastelijns/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/highline-1.7.10/lib/highline.rb:624: warning: Using the last argument as keyword parameters is deprecated
1. app/build/outputs/apk/androidTest/debug/app.apk
2. app/build/outputs/apk/debug/app.apk
?  2
[11:30:45]: To not be asked about this value, you can specify it using 'tests_apk_path'
[11:30:45]: Select your debug tests APK
/Users/timcastelijns/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/highline-1.7.10/lib/highline.rb:624: warning: Using the last argument as keyword parameters is deprecated
1. app/build/outputs/apk/androidTest/debug/app.apk
2. app/build/outputs/apk/debug/app.apk
?  1
[11:30:47]: Validating app APK
[11:30:47]: $ /Users/timcastelijns/Library/Android/sdk/build-tools/29.0.2/aapt dump permissions app/build/outputs/apk/debug/app.apk
[11:30:47]: ▸ package: com.xxx.debug
[11:30:47]: ▸ uses-permission: name='android.permission.DISABLE_KEYGUARD'
[11:30:47]: ▸ uses-permission: name='android.permission.WAKE_LOCK'
[11:30:47]: ▸ uses-permission: name='android.permission.WRITE_EXTERNAL_STORAGE'
[11:30:47]: ▸ uses-permission: name='android.permission.READ_EXTERNAL_STORAGE'
[11:30:47]: ▸ uses-permission: name='android.permission.CHANGE_CONFIGURATION'
[11:30:47]: ▸ uses-permission: name='android.permission.INTERNET'
[11:30:47]: ▸ uses-permission: name='android.permission.ACCESS_NETWORK_STATE'
[11:30:47]: ▸ uses-permission: name='android.permission.BLUETOOTH'
[11:30:47]: ▸ uses-permission: name='android.permission.BLUETOOTH_ADMIN'
[11:30:47]: ▸ uses-permission: name='android.permission.ACCESS_FINE_LOCATION'
[11:30:47]: ▸ uses-permission: name='android.permission.FOREGROUND_SERVICE'
[11:30:47]: ▸ uses-permission: name='android.permission.ACCESS_WIFI_STATE'
[11:30:47]: ▸ uses-permission: name='com.google.android.c2dm.permission.RECEIVE'
[11:30:47]: Installing app APK
[11:30:47]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 install -t -r app/build/outputs/apk/debug/app.apk
[11:30:47]: ▸ Performing Streamed Install
[11:30:50]: ▸ Success
[11:30:50]: Installing tests APK
[11:30:50]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 install -t -r app/build/outputs/apk/androidTest/debug/app.apk
[11:30:50]: ▸ Performing Streamed Install
[11:30:51]: ▸ Success
[11:30:51]: Granting the permission necessary to change locales on the device
[11:30:51]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell pm grant com.xxx.debug android.permission.CHANGE_CONFIGURATION
[11:30:51]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell getprop ro.build.version.sdk
[11:30:51]: ▸ 29
[11:30:51]: Granting the permissions necessary to access device external storage
[11:30:51]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell pm grant com.xxx.debug android.permission.WRITE_EXTERNAL_STORAGE
[11:30:51]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell pm grant com.xxx.debug android.permission.READ_EXTERNAL_STORAGE
[11:30:51]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell getprop ro.build.version.sdk
[11:30:51]: ▸ 29
[11:30:51]: $ /Users/timcastelijns/Library/Android/sdk/build-tools/29.0.2/aapt dump badging app/build/outputs/apk/debug/app.apk
[11:30:51]: ▸ package: name='com.xxx.debug' versionCode='5' versionName='0.9.1-dev-81e3' compileSdkVersion='29' compileSdkVersionCodename='10'
[11:30:51]: ▸ sdkVersion:'24'
[11:30:51]: ▸ targetSdkVersion:'29'
[11:30:51]: ▸ uses-permission: name='android.permission.DISABLE_KEYGUARD'
[11:30:51]: ▸ uses-permission: name='android.permission.WAKE_LOCK'
[11:30:51]: ▸ uses-permission: name='android.permission.WRITE_EXTERNAL_STORAGE'
[11:30:51]: ▸ uses-permission: name='android.permission.READ_EXTERNAL_STORAGE'
[11:30:51]: ▸ uses-permission: name='android.permission.CHANGE_CONFIGURATION'
[11:30:51]: ▸ uses-permission: name='android.permission.INTERNET'
[11:30:51]: ▸ uses-permission: name='android.permission.ACCESS_NETWORK_STATE'
[11:30:51]: ▸ uses-permission: name='android.permission.BLUETOOTH'
[11:30:51]: ▸ uses-permission: name='android.permission.BLUETOOTH_ADMIN'
[11:30:51]: ▸ uses-permission: name='android.permission.ACCESS_FINE_LOCATION'
[11:30:51]: ▸ uses-permission: name='android.permission.FOREGROUND_SERVICE'
[11:30:51]: ▸ uses-permission: name='android.permission.ACCESS_WIFI_STATE'
[11:30:51]: ▸ uses-permission: name='com.google.android.c2dm.permission.RECEIVE'
[11:30:51]: ▸ application-label:'xxx'
[11:30:51]: ▸ application-label-af:'xxx'
[11:30:51]: ▸ application-label-am:'xxx'
[11:30:51]: ▸ application-label-ar:'xxx'
[11:30:51]: ▸ application-label-as:'xxx'
[11:30:51]: ▸ application-label-az:'xxx'
[11:30:51]: ▸ application-label-be:'xxx'
[11:30:51]: ▸ application-label-bg:'xxx'
[11:30:51]: ▸ application-label-bn:'xxx'
[11:30:51]: ▸ application-label-bs:'xxx'
[11:30:51]: ▸ application-label-ca:'xxx'
[11:30:51]: ▸ application-label-cs:'xxx'
[11:30:51]: ▸ application-label-da:'xxx'
[11:30:51]: ▸ application-label-de:'xxx'
[11:30:51]: ▸ application-label-el:'xxx'
[11:30:51]: ▸ application-label-en-AU:'xxx'
[11:30:51]: ▸ application-label-en-CA:'xxx'
[11:30:51]: ▸ application-label-en-GB:'xxx'
[11:30:51]: ▸ application-label-en-IE:'xxx'
[11:30:51]: ▸ application-label-en-IN:'xxx'
[11:30:51]: ▸ application-label-en-XC:'[[xxx]]'
[11:30:51]: ▸ application-label-en-ZZ:'[[xxx]]'
[11:30:51]: ▸ application-label-es:'xxx'
[11:30:51]: ▸ application-label-es-CL:'xxx'
[11:30:51]: ▸ application-label-es-CO:'xxx'
[11:30:51]: ▸ application-label-es-CR:'xxx'
[11:30:51]: ▸ application-label-es-ES:'xxx'
[11:30:51]: ▸ application-label-es-MX:'xxx'
[11:30:51]: ▸ application-label-es-PA:'xxx'
[11:30:51]: ▸ application-label-es-PE:'xxx'
[11:30:51]: ▸ application-label-es-US:'xxx'
[11:30:51]: ▸ application-label-et:'xxx'
[11:30:51]: ▸ application-label-eu:'xxx'
[11:30:51]: ▸ application-label-fa:'xxx'
[11:30:51]: ▸ application-label-fi:'xxx'
[11:30:51]: ▸ application-label-fr:'xxx'
[11:30:51]: ▸ application-label-fr-CA:'xxx'
[11:30:51]: ▸ application-label-gl:'xxx'
[11:30:51]: ▸ application-label-gu:'xxx'
[11:30:51]: ▸ application-label-hi:'xxx'
[11:30:51]: ▸ application-label-hr:'xxx'
[11:30:51]: ▸ application-label-hu:'xxx'
[11:30:51]: ▸ application-label-hy:'xxx'
[11:30:51]: ▸ application-label-in:'xxx'
[11:30:51]: ▸ application-label-is:'xxx'
[11:30:51]: ▸ application-label-is-IS:'xxx'
[11:30:51]: ▸ application-label-it:'xxx'
[11:30:51]: ▸ application-label-iw:'xxx'
[11:30:51]: ▸ application-label-ja:'xxx'
[11:30:51]: ▸ application-label-ka:'xxx'
[11:30:51]: ▸ application-label-kk:'xxx'
[11:30:51]: ▸ application-label-km:'xxx'
[11:30:51]: ▸ application-label-kn:'xxx'
[11:30:51]: ▸ application-label-ko:'xxx'
[11:30:51]: ▸ application-label-ky:'xxx'
[11:30:51]: ▸ application-label-lo:'xxx'
[11:30:51]: ▸ application-label-lt:'xxx'
[11:30:51]: ▸ application-label-lv:'xxx'
[11:30:51]: ▸ application-label-mk:'xxx'
[11:30:51]: ▸ application-label-ml:'xxx'
[11:30:51]: ▸ application-label-mn:'xxx'
[11:30:51]: ▸ application-label-mr:'xxx'
[11:30:51]: ▸ application-label-ms:'xxx'
[11:30:51]: ▸ application-label-my:'xxx'
[11:30:51]: ▸ application-label-nb:'xxx'
[11:30:51]: ▸ application-label-ne:'xxx'
[11:30:51]: ▸ application-label-nl:'xxx'
[11:30:51]: ▸ application-label-no:'xxx'
[11:30:51]: ▸ application-label-or:'xxx'
[11:30:51]: ▸ application-label-pa:'xxx'
[11:30:51]: ▸ application-label-pl:'xxx'
[11:30:51]: ▸ application-label-pt:'xxx'
[11:30:51]: ▸ application-label-pt-BR:'xxx'
[11:30:51]: ▸ application-label-pt-PT:'xxx'
[11:30:51]: ▸ application-label-ro:'xxx'
[11:30:51]: ▸ application-label-ru:'xxx'
[11:30:51]: ▸ application-label-si:'xxx'
[11:30:51]: ▸ application-label-sk:'xxx'
[11:30:51]: ▸ application-label-sl:'xxx'
[11:30:51]: ▸ application-label-sq:'xxx'
[11:30:51]: ▸ application-label-sr:'xxx'
[11:30:51]: ▸ application-label-sr-Latn:'xxx'
[11:30:51]: ▸ application-label-sv:'xxx'
[11:30:51]: ▸ application-label-sw:'xxx'
[11:30:51]: ▸ application-label-ta:'xxx'
[11:30:51]: ▸ application-label-te:'xxx'
[11:30:51]: ▸ application-label-th:'xxx'
[11:30:51]: ▸ application-label-tl:'xxx'
[11:30:51]: ▸ application-label-tr:'xxx'
[11:30:51]: ▸ application-label-uk:'xxx'
[11:30:51]: ▸ application-label-ur:'xxx'
[11:30:51]: ▸ application-label-uz:'xxx'
[11:30:51]: ▸ application-label-vi:'xxx'
[11:30:51]: ▸ application-label-zh-CN:'xxx'
[11:30:51]: ▸ application-label-zh-HK:'xxx'
[11:30:51]: ▸ application-label-zh-TW:'xxx'
[11:30:51]: ▸ application-label-zu:'xxx'
[11:30:51]: ▸ application-icon-120:'res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application-icon-160:'res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application-icon-240:'res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application-icon-320:'res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application-icon-480:'res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application-icon-640:'res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application-icon-65534:'res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application: label='xxx' icon='res/mipmap-anydpi-v26/ic_launcher.xml'
[11:30:51]: ▸ application-debuggable
[11:30:51]: ▸ launchable-activity: name='com.xxx.MainActivity'  label='' icon=''
[11:30:51]: ▸ feature-group: label=''
[11:30:51]: ▸   uses-feature: name='android.hardware.bluetooth'
[11:30:51]: ▸   uses-implied-feature: name='android.hardware.bluetooth' reason='requested android.permission.BLUETOOTH permission, requested android.permission.BLUETOOTH_ADMIN permission, and targetSdkVersion > 4'
[11:30:51]: ▸   uses-feature: name='android.hardware.faketouch'
[11:30:51]: ▸   uses-implied-feature: name='android.hardware.faketouch' reason='default feature for all apps'
[11:30:51]: ▸   uses-feature: name='android.hardware.location'
[11:30:51]: ▸   uses-implied-feature: name='android.hardware.location' reason='requested android.permission.ACCESS_FINE_LOCATION permission'
[11:30:51]: ▸   uses-feature: name='android.hardware.screen.portrait'
[11:30:51]: ▸   uses-implied-feature: name='android.hardware.screen.portrait' reason='one or more activities have specified a portrait orientation'
[11:30:51]: ▸   uses-feature: name='android.hardware.wifi'
[11:30:51]: ▸   uses-implied-feature: name='android.hardware.wifi' reason='requested android.permission.ACCESS_WIFI_STATE permission'
[11:30:51]: ▸ main
[11:30:51]: ▸ other-activities
[11:30:51]: ▸ other-receivers
[11:30:51]: ▸ other-services
[11:30:51]: ▸ supports-screens: 'small' 'normal' 'large' 'xlarge'
[11:30:51]: ▸ supports-any-density: 'true'
[11:30:51]: ▸ locales: '--_--' 'af' 'am' 'ar' 'as' 'az' 'be' 'bg' 'bn' 'bs' 'ca' 'cs' 'da' 'de' 'el' 'en-AU' 'en-CA' 'en-GB' 'en-IE' 'en-IN' 'en-XC' 'en-ZZ' 'es' 'es-CL' 'es-CO' 'es-CR' 'es-ES' 'es-MX' 'es-PA' 'es-PE' 'es-US' 'et' 'eu' 'fa' 'fi' 'fr' 'fr-CA' 'gl' 'gu' 'hi' 'hr' 'hu' 'hy' 'in' 'is' 'is-IS' 'it' 'iw' 'ja' 'ka' 'kk' 'km' 'kn' 'ko' 'ky' 'lo' 'lt' 'lv' 'mk' 'ml' 'mn' 'mr' 'ms' 'my' 'nb' 'ne' 'nl' 'no' 'or' 'pa' 'pl' 'pt' 'pt-BR' 'pt-PT' 'ro' 'ru' 'si' 'sk' 'sl' 'sq' 'sr' 'sr-Latn' 'sv' 'sw' 'ta' 'te' 'th' 'tl' 'tr' 'uk' 'ur' 'uz' 'vi' 'zh-CN' 'zh-HK' 'zh-TW' 'zu'
[11:30:51]: ▸ densities: '120' '160' '240' '320' '480' '640' '65534'
[11:30:51]: ▸ native-code: 'arm64-v8a' 'armeabi' 'armeabi-v7a' 'mips' 'mips64' 'x86' 'x86_64'
[11:30:51]: Running tests for locale: en-US
[11:30:51]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell am instrument --no-window-animation -w \
-e testLocale en_US \
-e endingLocale en_US \
-e appendTimestamp true \
-e package com.xxx.screengrab \
com.xxx.debug.test/androidx.test.runner.AndroidJUnitRunner
[11:30:57]: ▸ com.xxx.screengrab.BasicScreenTest:.
[11:30:57]: ▸ Time: 2.788
[11:30:57]: ▸ OK (1 test)
[11:30:57]: Pulling captured screenshots for locale en-US from the device
[11:30:57]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 pull /sdcard/com.xxx.debug/screengrab /var/folders/9w/qkjs37fx0jb6h2jgf78k8nq00000gp/T/d20200219-60338-11jv2r6
ls: /data/data/com.xxx.debug/app_screengrab: Permission denied
[11:30:57]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/en-US/images/screenshots: Permission denied
[11:30:57]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/de-DE/images/screenshots: Permission denied
[11:30:57]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/nl-NL/images/screenshots: Permission denied
[11:30:57]: Exit status: 1
[11:30:57]: Screenshots copied to fastlane/metadata/android/en-US/images/en-US/images/phoneScreenshots
[11:30:57]: Running tests for locale: de-DE
[11:30:57]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell am instrument --no-window-animation -w \
-e testLocale de_DE \
-e endingLocale en_US \
-e appendTimestamp true \
-e package com.xxx.screengrab \
com.xxx.debug.test/androidx.test.runner.AndroidJUnitRunner
[11:31:05]: ▸ com.xxx.screengrab.BasicScreenTest:.
[11:31:05]: ▸ Time: 4.805
[11:31:05]: ▸ OK (1 test)
[11:31:05]: Pulling captured screenshots for locale de-DE from the device
[11:31:06]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 pull /sdcard/com.xxx.debug/screengrab /var/folders/9w/qkjs37fx0jb6h2jgf78k8nq00000gp/T/d20200219-60338-14t7pk7
ls: /data/data/com.xxx.debug/app_screengrab: Permission denied
[11:31:06]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/en-US/images/screenshots: Permission denied
[11:31:06]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/de-DE/images/screenshots: Permission denied
[11:31:06]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/nl-NL/images/screenshots: Permission denied
[11:31:06]: Exit status: 1
[11:31:06]: Screenshots copied to fastlane/metadata/android/de-DE/images/de-DE/images/phoneScreenshots
[11:31:06]: Screenshots copied to fastlane/metadata/android/en-US/images/de-DE/images/phoneScreenshots
[11:31:06]: Running tests for locale: nl-NL
[11:31:06]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 shell am instrument --no-window-animation -w \
-e testLocale nl_NL \
-e endingLocale en_US \
-e appendTimestamp true \
-e package com.xxx.screengrab \
com.xxx.debug.test/androidx.test.runner.AndroidJUnitRunner
[11:31:14]: ▸ com.xxx.screengrab.BasicScreenTest:.
[11:31:14]: ▸ Time: 3.641
[11:31:14]: ▸ OK (1 test)
[11:31:15]: Pulling captured screenshots for locale nl-NL from the device
[11:31:15]: $ /Users/timcastelijns/Library/Android/sdk/platform-tools/adb -s 8B5X140E5 pull /sdcard/com.xxx.debug/screengrab /var/folders/9w/qkjs37fx0jb6h2jgf78k8nq00000gp/T/d20200219-60338-1um9ldk
ls: /data/data/com.xxx.debug/app_screengrab: Permission denied
[11:31:15]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/en-US/images/screenshots: Permission denied
[11:31:15]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/de-DE/images/screenshots: Permission denied
[11:31:15]: Exit status: 1
ls: /data/user/0/com.xxx.debug/files/com.xxx.debug/screengrab/nl-NL/images/screenshots: Permission denied
[11:31:15]: Exit status: 1
[11:31:15]: Screenshots copied to fastlane/metadata/android/de-DE/images/nl-NL/images/phoneScreenshots
[11:31:15]: Screenshots copied to fastlane/metadata/android/nl-NL/images/nl-NL/images/phoneScreenshots
[11:31:15]: Screenshots copied to fastlane/metadata/android/en-US/images/nl-NL/images/phoneScreenshots
[11:31:15]: Generating HTML Report
[11:31:15]: Successfully created HTML file with an overview of all the screenshots: '/Users/timcastelijns/git/xxx/fastlane/metadata/android/screenshots.html'
[11:31:15]: Captured 6 new screenshots! 📷✨</pre>
</details>

### Environment

<details>
  <pre><details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.15.2                                     |
| Ruby                        | 2.7.0                                       |
| Bundler?                    | true                                        |
| Git                         | git version 2.17.1                          |
| Installation Source         | ~/.rbenv/versions/2.7.0/bin/fastlane        |
| Host                        | Mac OS X 10.15.2 (19C57)                    |
| Ruby Lib Dir                | ~/.rbenv/versions/2.7.0/lib                 |
| OpenSSL Version             | OpenSSL 1.1.1d  10 Sep 2019                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 11.3                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:android)

platform :android do

  desc "Build debug and test APK for screenshots"
  lane :build_for_screengrab do

    gradle(
      task: 'clean'
    )

    build_android_app(
      task: 'assemble',
      build_type: 'Debug'
    )

    build_android_app(
      task: 'assemble',
      build_type: 'AndroidTest'
    )

    copy_artifacts(
      artifacts: ['app/build/outputs/apk/debug/*.apk'],
      target_path: 'app/build/outputs/apk/debug/screengrab/'
    )

    copy_artifacts(
      artifacts: ['app/build/outputs/apk/androidTest/debug/*.apk'],
      target_path: 'app/build/outputs/apk/androidTest/debug/screengrab/'
    )

  end

  desc "Move/rename APKs for Screengrabfile"
  lane :rename_apks do

    sh("find ../app/build/outputs/apk/debug/screengrab -name '*.apk' -exec mv '{}' ../app/build/outputs/apk/debug/screengrab/app.apk \\;")
    sh("find ../app/build/outputs/apk/androidTest/debug/screengrab -name '*.apk' -exec mv '{}' ../app/build/outputs/apk/androidTest/debug/screengrab/app.apk \\;")

  end

  desc "Rebuild APKs and generate screenshots"
  lane :screenshots do

    build_for_screengrab

    #rename_apks

    screengrab

  end

end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
json_key_file("") # Path to the json secret file - Follow https://docs.fastlane.tools/actions/supply/#setup to get one
package_name("com.xxx.debug") # e.g. com.krausefx.app

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.142.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| did_you_mean              | 1.4.0        |
| bundler                   | 2.1.4        |
| uri                       | 0.10.0       |
| CFPropertyList            | 3.0.2        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| atomos                    | 0.1.3        |
| babosa                    | 1.0.3        |
| claide                    | 1.0.3        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.4.1        |
| unf_ext                   | 0.0.7.6      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.5        |
| emoji_regex               | 1.0.1        |
| excon                     | 0.72.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.17.3       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.13.1       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.1.0        |
| memoist                   | 0.16.2       |
| multi_json                | 1.14.1       |
| os                        | 1.0.1        |
| signet                    | 0.12.0       |
| googleauth                | 0.10.0       |
| httpclient                | 2.8.3        |
| mini_mime                 | 1.0.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.36.4       |
| google-cloud-env          | 1.3.0        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.25.1       |
| json                      | 2.3.0        |
| mini_magick               | 4.10.1       |
| multi_xml                 | 0.6.0        |
| plist                     | 3.5.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.6.1        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.15.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2020-02-19**
</details></pre>
</details>

**Comment by henningBunk** [comment]
Can confirm. same weird behaviour under 2.142.0 while 2.141.0 does work.

**Comment by joshdholtz** [comment]
👋  That is not right all 😱 I will look into right away and revert if not a quick fix. Thanks for reporting!

**Comment by andrzejchm** [comment]
I believe it might have been caused by this pr: 
https://github.com/fastlane/fastlane/pull/15994/files

**Comment by Jeehut** [comment]
@timc-mediamonks, @henningBunk & @andrzejchm: Could you guys please write which version of Android Studio you are using `screengrab` with? Cause when I implemented the change in #15994, it was still unclear why the issue happened in the first place.

My best guess there was that the behavior of Android or Android Studio tooling (`adb` maybe?) changed why it had failed for me. I'm using Android Studio 3.6 RC3 (RC2 back then when I implemented it) and the released version is outputting the files to paths like `metadata/android/de-DE/images/phoneScreenshots/*.png` just as expected.

**Comment by timc-mediamonks** [comment]
@Jeehut  using AS 3.5.3, the latest stable version.

**Comment by andrzejchm** [comment]
hey @Jeehut `3.5.3` is my android version. I've tested this on emulator api 29

**Comment by Jeehut** [comment]
In that case, could you test if an upgrade to Android Studio 3.6 RC3 fixes the issue? If yes, I think we need to update the code to behave differently depending on the Android Studio version installed.

**Comment by henningBunk** [comment]
no android studio, app is build on a ci server. sdk tools version 28 are used. target is also 28. emulator is also 28.

**Comment by andrzejchm** [comment]
tested on Android studio 3.6 rc3. same issue

**Comment by Jeehut** [comment]
Okay, then I have no idea why this is happening. Last guess I have is it's the `minSdkVersion`, @andrzejchm could you maybe try setting it to `28` and try again?

Regardless of the search for the cause, I have an idea how to fix it and just created a PR with: In theory, this should solve the issue for both cases. Feel free to try that branch and confirm/deny that it fixes the issue for you: https://github.com/fastlane/fastlane/pull/16063

**Comment by joshdholtz** [comment]
@Jeehut I think I found the issue! Its here 👉 https://github.com/fastlane/fastlane/blob/9035f306ab929b04839be1c13c9b2a4fe8c766b1/screengrab/lib/screengrab/runner.rb#L326

```ruby
screenshots_dir_pattern = File.join(pull_dir, '**', "screenshots")
```

I think this pattern needs to include `locale`? 🤔

**Comment by Jeehut** [comment]
@joshdholtz I don't understand. Isn't the `**` including everything in between `pull_dir` and `screenshots` already, including any locales?

**Comment by joshdholtz** [comment]
This will be fixed in #16094 💪  And will be out soon 😊
